src/pudl/transform/ferc1.py

```diff
@@ -853,6 +964,228 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+            .pipe(
+                self.assign_utility_id_ferc1,
+                source_ferc1=Ferc1Source.XBRL,
+            )
+        )
+
+    @cache_df(key="xbrl")
+    def merge_instant_and_duration_tables_xbrl(
+        self,
+        raw_xbrl_instant: pd.DataFrame,
+        raw_xbrl_duration: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Merge XBRL instant and duration tables, reshaping instant as needed.
+
+        FERC1 XBRL instant period signifies that it is true as of the reported date,
+        while a duration fact pertains to the specified time period. The ``date`` column
+        for an instant fact corresponds to the ``end_date`` column of a duration fact.
+
+        Note: This should always be applied before :meth:``rename_columns``
+
+        Args:
+            raw_xbrl_instant: table representing XBRL instant facts.
+            raw_xbrl_duration: table representing XBRL duration facts.
+
+        Returns:
+            A unified table combining the XBRL duration and instant facts, if both types
+            of facts were present. If either input dataframe is empty, the other
+            dataframe is returned unchanged, except that several unused columns are
+            dropped. If both input dataframes are empty, an empty dataframe is returned.
+        """
+        drop_cols = ["filing_name", "index"]
+        # Ignore errors in case not all drop_cols are present.
+        instant = raw_xbrl_instant.drop(columns=drop_cols, errors="ignore")
+        duration = raw_xbrl_duration.drop(columns=drop_cols, errors="ignore")
+
+        if instant.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL instant table found, returning the "
+                "duration table."
+            )
+            return duration
+        if duration.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL duration table found, returning "
+                "instant table."
+            )
+            return instant
+
+        instant_axes = [
+            col for col in raw_xbrl_instant.columns if col.endswith("_axis")
+        ]
+        duration_axes = [
+            col for col in raw_xbrl_duration.columns if col.endswith("_axis")
+        ]
+        if set(instant_axes) != set(duration_axes):
+            raise ValueError(
+                f"{self.table_id.value}: Instant and Duration XBRL Axes do not match.\n"
+                f"    instant: {instant_axes}\n"
+                f"    duration: {duration_axes}"
+            )
+
+        # Everything above here is the same as the parent function, but here we need
+        # to reshape the instant table such that end-of-last-year becomes
+        # beginning-of-this-year in a separate column. Probably going to need to factor
+        # out the shared vs. different code into separate methods once we know what this
+        # should look like.
+        instant = self.reshape_starting_ending_balances(instant)
```

Comment:
And another single-line reshaping intervention in an otherwise exact copy of the inherited method.

I could see having `reshape_instant_xbrl()` and `reshape_duration_xbrl()` methods that get called here but do nothing if reshaping isn't required. Is this a bad pattern to adopt?

Reply:
I've updated the inherited method to make this call, but have it do nothing unless the child class has implemented it for the tables that need reshaping. See `process_instant_xbrl()` and `process_duration_xbrl()`